diff --git a/Makefile b/Makefile
index a22eb5d..c55c8cf 100644
--- a/Makefile
+++ b/Makefile
@@ -383,6 +383,19 @@ else ifeq ($(platform), miyoo)
 	FLAGS += -fomit-frame-pointer -ffast-math -march=armv5te -mtune=arm926ej-s
 	fpic := -fPIC
 
+# target platform zero28 MagicX Zero28
+else ifeq ($(platform), zero28)
+	TARGET := $(TARGET_NAME)_libretro.so
+	CC = $(CROSS_COMPILE)gcc
+	CXX = $(CROSS_COMPILE)g++
+	AR = $(CROSS_COMPILE)ar
+	SHARED := -shared -Wl,--no-undefined
+	LDFLAGS += -shared -Wl,--version-script=link.T
+	CFLAGS += -DARM -DALIGN_DWORD -falign-functions=16 -pipe -fstack-protector
+	CFLAGS += -mtune=cortex-a53 -mcpu=cortex-a53 -march=armv8-a
+	fpic := -fPIC
+	HAVE_NEON = 1
+
 # Windows MSVC 2003 Xbox 1
 else ifeq ($(platform), xbox1_msvc2003)
 TARGET := $(TARGET_NAME)_libretro_xdk1.lib
