diff --git a/Makefile b/Makefile
index 67a7e1f..184ec33 100644
--- a/Makefile
+++ b/Makefile
@@ -326,6 +326,18 @@ else ifeq ($(platform), miyoo)
 	FLAGS += -fomit-frame-pointer -ffast-math -mcpu=arm926ej-s
 	fpic := -fPIC
 
+# target platform zero28 MagicX Zero28
+else ifeq ($(platform), zero28)
+	TARGET := $(TARGET_NAME)_libretro.so
+	CC = $(CROSS_COMPILE)gcc
+	CXX = $(CROSS_COMPILE)g++
+	AR = $(CROSS_COMPILE)ar
+	SHARED := -shared -Wl,--no-undefined
+	LDFLAGS += -shared -Wl,--version-script=link.T
+	FLAGS += -fomit-frame-pointer -ffast-math -mtune=cortex-a53 -mcpu=cortex-a53 -march=armv8-a
+	fpic := -fPIC
+	HAVE_NEON = 1
+
 # Raspberry Pi 1
 else ifeq ($(platform), rpi1)
 	TARGET := $(TARGET_NAME)_libretro.so
